ionoscloud/data_source_autoscaling_group.go

```diff
@@ -0,0 +1,300 @@
+							Computed:    true,
+							Type:        schema.TypeInt,
+							Description: "The amount of memory for the VMs in MB, e.g. 2048. Size must be specified in multiples of 256 MB with a minimum of 256 MB; however, if you set ramHotPlug to TRUE then you must use a minimum of 1024 MB. If you set the RAM size more than 240GB, then ramHotPlug will be set to FALSE and can not be set to TRUE unless RAM size not set to less than 240GB.",
+						},
+						"volume": {
+							Type:        schema.TypeList,
+							Description: "List of volumes associated with this Replica. Only a single volume is currently supported.",
+							Computed:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"image": {
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: "The image installed on the volume. Only the UUID of the image is presently supported.",
+									},
+									"name": {
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: "Name for this replica volume.",
+									},
+									"size": {
+										Computed:    true,
+										Type:        schema.TypeInt,
+										Description: "User-defined size for this replica volume in GB.",
+									},
+									"ssh_keys": {
+										Type:     schema.TypeList,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Computed: true,
+									},
+									"type": {
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: "Storage Type for this replica volume. Possible values: SSD, HDD, SSD_STANDARD or SSD_PREMIUM",
+									},
+									"user_data": {
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: "user-data (Cloud Init) for this replica volume.",
+									},
+								}},
+						},
+					},
+				},
+			},
+			"datacenter_id": {
+				Type:        schema.TypeString,
+				Description: "Unique identifier for the resource",
+				Computed:    true,
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Description: "Location of the data center.",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func dataSourceAutoscalingGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).AutoscalingClient
+
+	id, idOk := d.GetOk("id")
+	name, nameOk := d.GetOk("name")
+
+	if idOk && nameOk {
+		diags := diag.FromErr(fmt.Errorf("id and name cannot be both specified at the same time"))
+		return diags
+	}
+	if !idOk && !nameOk {
+		diags := diag.FromErr(fmt.Errorf("please provide either the group id or name"))
+		return diags
+	}
+
+	var group autoscaling.Group
+	var err error
+
+	if idOk {
+		group, _, err = client.GetGroup(ctx, id.(string), 2)
+		if err != nil {
+			diags := diag.FromErr(fmt.Errorf("an error occurred while fetching group with ID %s: %w", id.(string), err))
```

Comment:
can return `diag.FromErr` directly without the assignment

Reply:
@cbeti-ionos solved in c9df22e0066ca69a7809de59cd124fe97fec9580